lp-core-platform/lp-cp-apis/src/main/java/eu/learnpad/core/impl/qm/XwikiController.java

```diff
@@ -19,22 +19,33 @@
  */
 package eu.learnpad.core.impl.qm;
 
+import org.xwiki.component.annotation.Component;
+import org.xwiki.rest.XWikiRestComponent;
+
 import eu.learnpad.exception.impl.LpRestExceptionImpl;
+import eu.learnpad.qm.BridgeInterface;
 import eu.learnpad.qm.Controller;
 
-public class XwikiController extends Controller{
+@Component
```

Comment:
Same here, only empty constructor is useful here.  Moreover, to Inject components, you should not use the `new`.  There is a few ways to do that.

### `@Inject`
When you define your class attribute, you annotate it with an `@Inject` and then, at instantiation of the class, the attribute will automatically be filled with an instance of the corresponding component.

```
class MyController {
  @Inject
  MyBridge myBridge;
}
```


However, in our case, the class attribute is in a inherited class, therefore, first solution is not really feasible.  The following propositions may then be used.

### Provider
Instead of using `new` (which be likely break at some points for components), we'll use a provider.  This provider is like a factory.  and we should not do that in the constructor (because it's not supported by the components system of XWiki).
```
@Inject
Provider<MyBridge> myBridgeProvider;
```

And then, we could provide our attribute in a `initialize()` method.
```
void initialize() {
this.myBridge = myBridgeProvider.get();
```

I've never do that but it's on hints of a few of my colleagues :-)